src/joystick/n3ds/SDL_sysjoystick.c

```diff
@@ -0,0 +1,306 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2022 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+#include "../../SDL_internal.h"
+
+#ifdef SDL_JOYSTICK_N3DS
+
+/* This is the N3DS implementation of the SDL joystick API */
+
+#include <3ds.h>
+
+#include "../SDL_sysjoystick.h"
+#include "SDL_events.h"
+
+#define NB_BUTTONS 23
+
+/*
+  N3DS sticks values are roughly within +/-160
+  which is too small to pass the jitter tolerance.
+  This correction factor is applied to axis values
+  so they fit better in SDL's value range.
+*/
+#define CORRECTION_FACTOR_X SDL_JOYSTICK_AXIS_MAX / 160
```

Comment:
should maybe be put into parenthesis

Reply:
Having no parenthesis allows for slightly better precision without using floats. But this difference in precision is a difference of <1%. Do you think it is worth changing ?

Reply:
I went ahead and converted this to a macro in [d567803](https://github.com/libsdl-org/SDL/pull/6251/commits/d56780347a9d20233f330e2baf94a7d467eef279) to avoid the ambiguity with the order of operations.